frontstage/views/account/accept_transfer_survey.py

```diff
@@ -0,0 +1,119 @@
+import logging
+
+from flask import render_template, abort, request, url_for
+from structlog import wrap_logger
+from werkzeug.utils import redirect
+
+from frontstage import app
+from frontstage.common.authorisation import jwt_authorization
+from frontstage.controllers import party_controller, survey_controller
+from frontstage.controllers.party_controller import get_business_by_id
+from frontstage.exceptions.exceptions import ApiError
+from frontstage.views.account import account_bp
+
+logger = wrap_logger(logging.getLogger(__name__))
+
+
+@account_bp.route('/transfer-surveys/accept-transfer-surveys/<token>', methods=['GET'])
+def get_transfer_survey_summary(token):
+    """
+    Endpoint to verify transfer token and retrieve the summary page
+    :param token: transfer survey token
+    :type token: str
+    """
+    logger.info('Getting transfer survey summary', token=token)
+    try:
+        response = party_controller.verify_pending_survey_token(token)
+        pending_transfer_surveys = response.json()
+        transfer_dict = {}
+        distinct_businesses = set()
+        batch_number = pending_transfer_surveys[0]['batch_no']
+        originator_party = party_controller.get_respondent_party_by_id(pending_transfer_surveys[0]['shared_by'])
+        transferred_by = originator_party['emailAddress']
+        for pending_transfer_survey in pending_transfer_surveys:
+            distinct_businesses.add(pending_transfer_survey['business_id'])
+        for business_id in distinct_businesses:
+            business_surveys = []
+            for pending_transfer_survey in pending_transfer_surveys:
+                if pending_transfer_survey['business_id'] == business_id:
+                    business_surveys.append(survey_controller.get_survey(app.config['SURVEY_URL'],
+                                                                         app.config['BASIC_AUTH'],
+                                                                         pending_transfer_survey['survey_id']))
+            selected_business = get_business_by_id(business_id)
+            transfer_dict[selected_business[0]['name']] = {
+                'trading_as': selected_business[0]['trading_as'],
+                'surveys': business_surveys
+            }
+        return render_template('surveys/surveys-transfer/summary.html',
+                               transfer_dict=transfer_dict,
+                               batch_no=batch_number,
+                               transferred_by=transferred_by)
+
+    except ApiError as exc:
+        # Handle api errors
+        if exc.status_code == 409:
+            logger.info('Expired transfer survey email verification token', token=token, api_url=exc.url,
+                        api_status_code=exc.status_code)
+            abort(409)
```

Comment:
How are these status codes handled? In other flows (e.g. new account registration) we show a screen telling them "hey your link expired, tough beans". I have a feeling this will just show them our generic 500, or possibly this might be the first time we've ever just shown a straight up 404/409 error. Either way, I'd rather we rendered a small page to tell them what went wrong.

Reply:
Gareth has been made aware of this.. as there were no scenario suggested in the story the process is getting abort for now.. The suggesting you made is absolutely valid and will be played as the iteration story.